core/translate/expr.rs

```diff
@@ -967,6 +967,56 @@ pub fn translate_expr(
 
                             Ok(target_register)
                         }
+                        ScalarFunc::Iif => {
+                            let args = match args {
+                                Some(args) if args.len() == 3 => args,
+                                _ => crate::bail_parse_error!(
+                                    "{} requires exactly 3 arguments",
+                                    srf.to_string()
+                                ),
+                            };
+                            let temp_reg = program.alloc_register();
+                            translate_expr(
+                                program,
+                                referenced_tables,
+                                &args[0],
+                                temp_reg,
+                                precomputed_exprs_to_registers,
+                            )?;
+                            let jump_target_when_false = program.allocate_label();
+                            program.emit_insn_with_label_dependency(
+                                Insn::IfNot {
+                                    reg: temp_reg,
+                                    target_pc: jump_target_when_false,
+                                    null_reg: 1,
+                                },
+                                jump_target_when_false,
+                            );
+                            translate_expr(
+                                program,
+                                referenced_tables,
+                                &args[1],
+                                target_register,
+                                precomputed_exprs_to_registers,
+                            )?;
+                            let jump_target_result = program.allocate_label();
+                            program.emit_insn_with_label_dependency(
+                                Insn::Goto {
+                                    target_pc: jump_target_result,
+                                },
+                                jump_target_result,
+                            );
+                            program.preassign_label_to_next_insn(jump_target_when_false);
```

Comment:
you can just use `program.resolve_label(jump_target_when_false, program.offset())` here and below